internals/updates/update.go

```diff
@@ -0,0 +1,48 @@
+package updates
+
+import (
+	"context"
+	"time"
+
+	"github.com/rs/zerolog/log"
+)
+
+type GhostUpdaterCleaner interface {
+	Clean(ctx context.Context) error
+	UpdateID() int
```

Comment:
passing it to the cleaner makes it look like they will clean the updater for this specific id, when it cleans all ghost updaters.

Reply:
That was my original implementation: https://github.com/portainer/agent/pull/437/commits/9a1f71771a61ef836eb95715534c87c759430d9f#

Reply:
sorry, you are right. I made this comment https://github.com/portainer/agent/pull/437#discussion_r1126349216

just didn't see that it's actually not required. it's actually not really required for the Remove fn. we can and maybe should look for those ghosts also when agent is not an updated agent

Reply:
It could be a good idea. I am simply curious about the possibility of a new agent (not an updated agent) starting while the ghost updater container exists.